File: 0819-most-common-word/0819-most-common-word.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>
#include <algorithm>
#include <regex>
#include <cctype>

using namespace std;

class Solution {
public:
    string mostCommonWord(string paragraph, vector<string>& banned) {
        unordered_set<string> bannedSet(banned.begin(), banned.end());
        
        transform(paragraph.begin(), paragraph.end(), paragraph.begin(), ::tolower);
        regex word_regex(R"([a-z]+)");
        sregex_iterator iter(paragraph.begin(), paragraph.end(), word_regex);
        sregex_iterator end;

        unordered_map<string, int> wordCount;

        while (iter != end) {
            string word = iter->str();
            if (bannedSet.find(word) == bannedSet.end()) {
                wordCount[word]++;
            }
            ++iter;
        }

        string answer;
        int maxCount = 0;
        for (const auto& entry : wordCount) {
            if (entry.second > maxCount) {
                maxCount = entry.second;
                answer = entry.first;
            }
        }

        return answer;
    }
};

File: 0819-most-common-word/0819-most-common-word.py
import re
class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        # words = paragraph.split(' ')
        words = re.split(r'[ ,!:\s]+', paragraph)
        map = {}
        for i in range(len(words)):
            word = re.sub(r'[^a-zA-Z\s]', '', words[i])
            word = word.lower()
            if word not in banned:
                if word not in map.keys():
                    map[word] = 1
                else:
                    map[word] += 1
        max_rep = 0
        answer = ''
        for k in map.keys():
            if map[k] > max_rep:
                answer = k
                max_rep = map[k]
        return answer
                

                